frontend/src/worksheets/CreateWorkSheetForm.js

```diff
@@ -129,6 +133,15 @@ function CreateWorkSheetForm({
                     <Button
                       text="Mentés"
                       type="submit"
+                      onClick={() => setFinalize(false)}
+                      moreClassName="h-auto"
+                    />
+                  </div>
+                  <div className="buttons">
+                    <Button
+                      text="Mentés és készre jelentés"
+                      type="submit"
+                      onClick={() => setFinalize(true)}
```

Comment:
Ebben nem lehetsz biztos, hogy hamarabb lefut a state update, mint az onSubmit. Nem egyszerű a problémakör, van erről egy issue is a Formik repoban: https://github.com/formium/formik/issues/214
Talán eggyel jobb megoldás lehetne, ha nem state lenne a finalize, hanem ref, mert az instant megváltozik és biztos, hogy az onSubmit-re is korrekt lesz az értéke.
B variáció, de más UI/UX: lenne egy állapotválasztó select vagy radio group és a status is egy Formik-os field lenne végülis.